src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -287,18 +287,20 @@ private void OnGetHeadersMessageReceived(GetBlockByIndexPayload payload)
         private void OnInventoryReceived(IInventory inventory)
         {
             pendingKnownHashes.Remove(inventory.Hash);
-            knownHashes.Add(inventory.Hash);
-            system.TaskManager.Tell(inventory);
-            system.Blockchain.Tell(inventory, ActorRefs.NoSender);
             switch (inventory)
             {
                 case Transaction transaction:
+                    knownHashes.Add(inventory.Hash);
                     system.Consensus?.Tell(transaction);
                     break;
                 case Block block:
+                    if (block.Index > Blockchain.Singleton.Height + 500) return;
+                    knownHashes.Add(inventory.Hash);
                     UpdateLastBlockIndex(block.Index, false);
                     break;
             }
+            system.TaskManager.Tell(inventory);
+            system.Blockchain.Tell(inventory, ActorRefs.NoSender);
```

Comment:
Here it is `ActorRefs.NoSender`, so you can't use `Sender` in `Blockchain`.

Reply:
It was moved from line 292